[Flaky] spec/system/consumer/split_checkout_tax_not_incl_spec.rb

### What we should change and why (this is tech debt)
<!-- The use of X changed and part y is now redundant. We can remove it and... -->
```
Failures:

  1) As a consumer, I want to see adjustment breakdown a not-included tax for a customer with shipping address within the tax zone on legacy checkout will be charged tax on the order
     Failure/Error: expect(order_within_zone.additional_tax_total).to eq(1.3)
     
       expected: 1.3
            got: 0.0
     
       (compared using ==)
     
     [Screenshot Image]: /home/runner/work/openfoodnetwork/openfoodnetwork/tmp/capybara/screenshots/failures_r_spec_example_groups_as_a_consumer_i_want_to_see_adjustment_breakdown_a_not_included_tax_for_a_customer_with_shipping_address_within_the_tax_zone_on_legacy_checkout_will_be_charged_tax_on_the_order_216.png

     
     # ./spec/system/consumer/split_checkout_tax_not_incl_spec.rb:111:in `block (5 levels) in <main>'
     # ./spec/system/support/cuprite_setup.rb:41:in `block (2 levels) in <top (required)>'

Finished in 2 minutes 29.9 seconds
26 examples, 1 failure

Failed examples:

rspec ./spec/system/consumer/split_checkout_tax_not_incl_spec.rb:97 # As a consumer, I want to see adjustment breakdown a not-included tax for a customer with shipping address within the tax zone on legacy checkout will be charged tax on the order
I, [2023-06-12T14:37:35.916600 #2976]  INFO -- : [knapsack_pro] To retry all the tests assigned to this CI node, please run the following command on your machine:
I, [2023-06-12T14:37:35.916682 #2976]  INFO -- : [knapsack_pro] bundle exec rspec --format progress --default-path spec "spec/system/consumer/split_checkout_tax_not_incl_spec.rb" "spec/system/consumer/shopping/orders_spec.rb" "spec/system/consumer/shopping/checkout_paypal_spec.rb" "spec/system/consumer/groups_spec.rb" "spec/system/consumer/account/payments_spec.rb"
Stopped processing SimpleCov as a previous error not related to SimpleCov has been detected
Coverage report generated for RSpec to /home/runner/work/openfoodnetwork/openfoodnetwork/coverage. 7516 / 24211 LOC (31.04%) covered.
Error: Process completed with exit code 1.
```

### Context
<!-- This came up in PR #3451#issuecomment-461616636. -->
https://github.com/openfoodfoundation/openfoodnetwork/actions/runs/5244916759/jobs/9471599216


### Impact and timeline
<!-- Example: This simplifies the upgrade to Spree 3 and should be done after upgrading to Spree 2. -->

--- a/spec/support/split_checkout_helper.rb
+++ b/spec/support/split_checkout_helper.rb
@@ -43,6 +43,13 @@ module SplitCheckoutHelper
     expect(page).to have_button("Next - Order summary")
   end
 
+  def apply_voucher(code)
+    expect(page).to have_content "Apply voucher"
+    fill_in "Enter voucher code", with: code
+    click_button "Apply"
+    expect(page).to have_link "Remove code"
+  end
+
   def expect_to_be_on_first_step
     expect(page).to have_content("1 - Your details")
     expect(page).to have_selector("div.checkout-tab.selected", text: "1 - Your details")

--- a/spec/system/consumer/split_checkout_spec.rb
+++ b/spec/system/consumer/split_checkout_spec.rb
@@ -706,8 +706,7 @@ describe "As a consumer, I want to checkout my order" do
 
             shared_examples "adding voucher to the order" do
               before do
-                fill_in "Enter voucher code", with: "some_code"
-                click_button("Apply")
+                apply_voucher "some_code"
               end
 
               it "adds a voucher to the order" do
@@ -727,8 +726,7 @@ describe "As a consumer, I want to checkout my order" do
               it_behaves_like "adding voucher to the order"
 
               it "shows a warning message" do
-                fill_in "Enter voucher code", with: "some_code"
-                click_button("Apply")
+                apply_voucher "some_code"
 
                 expect(page).to have_content(
                   "Note: if your order total is less than your voucher " \
@@ -737,8 +735,7 @@ describe "As a consumer, I want to checkout my order" do
               end
 
               it "proceeds without requiring payment" do
-                fill_in "Enter voucher code", with: "some_code"
-                click_button("Apply")
+                apply_voucher "some_code"
 
                 expect(page).to have_content "No payment required"
                 click_button "Next - Order summary"
@@ -779,9 +776,7 @@ describe "As a consumer, I want to checkout my order" do
             end
 
             it "can re-enter a voucher" do
-              # Re-enter a voucher code
-              fill_in "Enter voucher code", with: "some_code"
-              click_button("Apply")
+              apply_voucher "some_code"
 
               expect(page).to have_content("$15.00 Voucher")
               expect(order.reload.voucher_adjustments.length).to eq(1)

--- a/spec/system/consumer/split_checkout_spec.rb
+++ b/spec/system/consumer/split_checkout_spec.rb
@@ -704,18 +704,12 @@ describe "As a consumer, I want to checkout my order" do
               visit checkout_step_path(:payment)
             end
 
-            shared_examples "adding voucher to the order" do
-              before do
-                apply_voucher "some_code"
-              end
+            it "adds a voucher to the order" do
+              apply_voucher "some_code"
 
-              it "adds a voucher to the order" do
-                expect(page).to have_content("$15.00 Voucher")
-                expect(order.reload.voucher_adjustments.length).to eq(1)
-              end
+              expect(page).to have_content "$15.00 Voucher"
+              expect(order.reload.voucher_adjustments.length).to eq(1)
             end
-
-            it_behaves_like "adding voucher to the order"
 
             context "when voucher covers more then the order total" do
               before do
@@ -723,19 +717,14 @@ describe "As a consumer, I want to checkout my order" do
                 order.save!
               end
 
-              it_behaves_like "adding voucher to the order"
-
-              it "shows a warning message" do
+              it "shows a warning message and doesn't require payment" do
                 apply_voucher "some_code"
 
+                expect(page).to have_content "$15.00 Voucher"
                 expect(page).to have_content(
                   "Note: if your order total is less than your voucher " \
                   "you may not be able to spend the remaining value."
                 )
-              end
-
-              it "proceeds without requiring payment" do
-                apply_voucher "some_code"
 
                 expect(page).to have_content "No payment required"
                 click_button "Next - Order summary"

--- a/spec/system/consumer/split_checkout_spec.rb
+++ b/spec/system/consumer/split_checkout_spec.rb
@@ -699,12 +699,6 @@ describe "As a consumer, I want to checkout my order" do
             create(:voucher_flat_rate, code: 'some_code', enterprise: distributor, amount: 15)
           end
 
-          it "shows voucher input" do
-            visit checkout_step_path(:payment)
-            expect(page).to have_field "Enter voucher code"
-            expect(page).to have_content "Apply voucher"
-          end
-
           describe "adding voucher to the order" do
             before do
               visit checkout_step_path(:payment)

--- a/spec/system/consumer/split_checkout_tax_incl_spec.rb
+++ b/spec/system/consumer/split_checkout_tax_incl_spec.rb
@@ -122,10 +122,7 @@ describe "As a consumer, I want to see adjustment breakdown" do
 
           click_button "Next - Payment method"
 
-          # add Voucher
-          fill_in "Enter voucher code", with: "some_code"
-          click_button("Apply")
-          expect(page).to have_link "Remove code"
+          apply_voucher "some_code"
 
           # Choose payment
           click_on "Next - Order summary"
@@ -155,9 +152,7 @@ describe "As a consumer, I want to see adjustment breakdown" do
             visit checkout_step_path(:details)
             proceed_to_payment
 
-            # add Voucher
-            fill_in "Enter voucher code", with: "good_code"
-            click_button("Apply")
+            apply_voucher "good_code"
 
             proceed_to_summary
 

--- a/spec/system/consumer/split_checkout_tax_not_incl_spec.rb
+++ b/spec/system/consumer/split_checkout_tax_not_incl_spec.rb
@@ -128,10 +128,7 @@ describe "As a consumer, I want to see adjustment breakdown" do
 
           click_button "Next - Payment method"
 
-          # add Voucher
-          fill_in "Enter voucher code", with: "some_code"
-          click_button("Apply")
-          expect(page).to have_selector ".voucher-added"
+          apply_voucher "some_code"
 
           click_on "Next - Order summary"
           click_on "Complete order"
@@ -163,9 +160,7 @@ describe "As a consumer, I want to see adjustment breakdown" do
             visit checkout_step_path(:details)
             proceed_to_payment
 
-            # add Voucher
-            fill_in "Enter voucher code", with: voucher.code
-            click_button("Apply")
+            apply_voucher voucher.code
 
             proceed_to_summary
             assert_db_voucher_adjustment(-2.00, -0.26)

